atat/src/ingress_manager.rs

```diff
@@ -387,8 +455,9 @@ where
                         self.buf_incomplete,
                         self.buf.len(),
                     );
-                    self.buf_incomplete = !self.buf.ends_with(self.line_term_char as char)
-                        && !self.buf.ends_with(self.format_char as char);
+                    self.buf_incomplete = self.buf.len() > 0
+                        && self.buf.get(self.buf.len() - 1) != Some(&self.line_term_char)
+                        && self.buf.get(self.buf.len() - 1) != Some(&self.format_char);
```

Comment:
Won't this change the logic slightly? If the buf is empty, the previous code will result in `true` while the new code will result in `false`.

(The absence of `starts_with` and `ends_with` is one of the downsides of using a `Vec`... Maybe those methods could be added to heapless? Should be fairly easy to implement.)

Reply:
Thats a great catch! Didn't even notice that myself.

> Maybe those methods could be added to heapless? Should be fairly easy to implement

That would be awesome, but until then, i'll fix the logic here without.

Reply:
https://github.com/japaric/heapless/pull/152